cost-analyzer/templates/_helpers.tpl

```diff
@@ -217,19 +215,17 @@ support templating a chart which uses the lookup function.
 {{- end -}}
 
 {{/*
-Verify the federated storage config secret exists with the expected key when cloud integration is enabled.
-Skip the check if CI/CD is enabled and skipSanityChecks is set. Argo CD, for example, does not
-support templating a chart which uses the lookup function.
+Verify the federated storage config secret exists with the expected key.
+Skip the check if CI/CD is enabled and skipSanityChecks is set. Argo CD, for
+example, does not support templating a chart which uses the lookup function.
 */}}
 {{- define "federatedStorageConfigSecretCheck" -}}
 {{- if (.Values.kubecostModel).federatedStorageConfigSecret }}
 {{- if not (and .Values.global.platforms.cicd.enabled .Values.global.platforms.cicd.skipSanityChecks) }}
 {{-  if .Capabilities.APIVersions.Has "v1/Secret" }}
   {{- $secret := lookup "v1" "Secret" .Release.Namespace .Values.kubecostModel.federatedStorageConfigSecret }}
   {{- if or (not $secret) (not (index $secret.data "federated-store.yaml")) }}
-    {{- if not .Values.kubecostModel.federatedStorageConfig }}
```

Comment:
I don't think we needed this extra conditional. This check only runs if the user has defined a `federatedStorageConfigSecret`.